File: DSA/TEST_CASES/GO_Lang_TestCases/SearchInsert.go
package main

import "fmt"

func searchInsert(nums []int, target int) int {
	if nums[len(nums)-1] < target {
		return len(nums)
	}
	for i := 0; i < len(nums); i++ {
		if nums[i] == target {
			return i
		}
		if nums[i] > target {
			return i
		}
	}
	return -1
}

func main() {
	L := []int{1, 2, 4, 5, 7, 8}
	// there are 6 elements in an list (0-5)

	fmt.Println(searchInsert(L, 3))
	fmt.Println(searchInsert(L, 4))
	fmt.Println(searchInsert(L, 5))
	fmt.Println(searchInsert(L, 33))
}

File: DSA/TEST_CASES/Python_TestCases/SearchInsert.py
class Solution:
    def searchInsert(self, nums, target):
        idx = 0
        for i in range(len(nums)):
            if nums[i] == target:
                return i
            if nums[i] < target:
                idx = i + 1
            else:
                break
                
        return idx

# a = [1,3,5,6]
# t = 5
# O = Solution()
# print(O.searchInsert(a, t))

print(O.searchInsert([1,2,6,7,8], 3))
print(O.searchInsert([1,2,6,7,8,9], 6))
print(O.searchInsert([1,2,6,7,8,10], 9))
